CRM/Contribute/Form/ContributionBase.php

```diff
@@ -293,7 +293,7 @@ public function preProcess() {
       $isPayLater = CRM_Utils_Array::value('is_pay_later', $this->_values);
 
       if ($isMonetary &&
-        (!$isPayLater || !empty($this->_values['payment_processor']))
+        (!$isPayLater && !empty($this->_values['payment_processor']))
```

Comment:
Yeah, I think we are populating $this->_value['payment_processor']  for pay_later -  & using the manual processor to provide pieces of logic rather than lots of if ($payLater) THEN type logic.

I introduced that in 4.7 to deal with some specific issues and because I feel that as a direction it provides potential for more simplicity